### ISSUE TITLE
different machine(v100 and p100) give differnt training result for deep learning models

### ISSUE THREAD

**Issue by world2vec** [googlecolab/colabtools#1635]
Hi,
I use torch==1.6.0+cu101 torchvision==0.7.0+cu101.
First My model can generated same result for different v100 machines.
But on P100 I found the result score is much lower.
And I confirmed the first epoch loss is different on V100 and P100.
What caused this difference? What should I check?
Thanks.

**Comment by world2vec** [comment]
Hi,
Any update? 
I can not control what type GPU I can get. In order to make sure I can reproduce my work, I can only run my experiment when I get a V100 GPU.

**Comment by world2vec** [comment]
This is annoying!
If I get P100 for sometime then I will always get V100, but I need P100 to reproduce my work.
So could you give some solution for this problem?
Thanks.

**Comment by hroser** [comment]
Hi, I have the same problem, could you find a solution or a reason for that?

Training my model with P100 takes longer (which would not be a problem) but also gives significantly lower model performance, which makes my experiments not reproducable if changed from V100 to P100.
Results on V100 are way better. Could mixed precision be a reason? Perhaps this is implicitly or automatically applied when using V100?

Would be very happy for any advice.

**Comment by hroser** [comment]
I am using Tensorflow 1.15.2, Keras 2.3.1

**Comment by hroser** [comment]
Found this, could perhaps help

https://github.com/YeongHyeon/GANomaly-PyTorch/issues/1

**Comment by world2vec** [comment]
This is still a problem. When colab force to allocate a V100 to me, my model performance(5 fold) has 10 percent degradation compared to with P100, this means I need re tuning all hyper parameters.
